notes/stv/20221014-test-benchmarker-01.txt

```diff
@@ -0,0 +1,325 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2022, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+
+
+
+    Target:
+
+        Test Spectra notebook & public examples before doing a PR (2022/10/13)
+
+    Result:
+
+        Success (*) 
+         (*) Benchmark runs but HDBSCAN fails to import
+
+
+# -----------------------------------------------------
+# Create a container to work with.
+#[user@desktop]
+
+    source "${HOME:?}/aglais.env"
+
+    agcolour=red
+    agproxymap=3000:3000
+
+    clientname=ansibler-${agcolour}
+    cloudname=iris-gaia-${agcolour}
+    configname=zeppelin-26.43-spark-6.26.43
+
+    podman run \
+        --rm \
+        --tty \
+        --interactive \
+        --name     "${clientname:?}" \
+        --hostname "${clientname:?}" \
+        --publish  "${agproxymap:?}" \
+        --env "cloudname=${cloudname:?}" \
+        --env "configname=${configname:?}" \
+        --env "SSH_AUTH_SOCK=/mnt/ssh_auth_sock" \
+        --volume "${SSH_AUTH_SOCK:?}:/mnt/ssh_auth_sock:rw,z" \
+        --volume "${HOME:?}/clouds.yaml:/etc/openstack/clouds.yaml:ro,z" \
+        --volume "${AGLAIS_CODE:?}/deployments:/deployments:ro,z" \
+        ghcr.io/wfau/atolmis/ansible-client:2022.07.25 \
+        bash
+
+
+# -----------------------------------------------------
+# Deploy everything.
+#[root@ansibler]
+
+    time \
+	source /deployments/hadoop-yarn/bin/deploy.sh
+
+
+	> real	77m19.087s
+	> user	10m18.075s
+	> sys	2m45.715s
+
+
+# -----------------------------------------------------
+# Import our live users.
+#[root@ansibler]
+
+    source /deployments/zeppelin/bin/create-user-tools.sh
+
+    import-test-users
+
+# -----------------------------------------------------
+# Start NGINX so that we can access using public IP.
+#[fedorat@zeppelin]
+
+    sudo service nginx start
+
```

Comment:
Does this mean that by default port 80 is open in the firewall ?

Reply:
Why are we leaving port 80 open and why is Nginx listening on it ?

Reply:
I may be wrong but if we close port 80 and someone tries to access our service via http:// wouldn't they then get a HTTP error response? Where if NGINX listens to port 80 and 443, if someone accesses port 80 they get redirected to 443, so that both http & https work?

Reply:
The other reason is that if we close 80 and only have 443 open, we'd have to setup SSL for all the test domains and modify our scripts so that both test & prod deploys set up SSL. Which is a good plan perhaps, but maybe needs to be done in a separate branch?

Reply:
>  ... we'd have to setup SSL for all the test domains and Which is a good plan perhaps, but maybe needs to be done in a separate branch?

Agreed, we don't need to solve this here.
See #838.